Kernel/Net/IPv4Socket.cpp

```diff
@@ -693,10 +694,55 @@ ErrorOr<void> IPv4Socket::ioctl(OpenFileDescription&, unsigned request, Userspac
         ifreq ifr;
         TRY(copy_from_user(&ifr, user_ifr));
 
+        if (request == SIOCGIFNAME) {
+            // NOTE: Network devices are 1-indexed since index 0 denotes an invalid device
+            if (ifr.ifr_index == 0)
+                return EINVAL;
+
+            size_t index = 1;
+            Optional<StringView> result {};
+
+            NetworkingManagement::the().for_each([&ifr, &index, &result](auto& adapter) {
```

Comment:
This kind of usage is sad to see, but nothing we can do about it without a refactor which I will not suggest to do in one PR with these changes being included.